src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle',
+  color: 'red' | 'green' | 'blue',
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: "triangle" = "triangle";
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const ab = a + b;
+    const bc = b + c;
+    const ac = a + c;
```

Comment:
The variables `a`, `b`, and `c` are not defined within the constructor. You should use `this.a`, `this.b`, and `this.c` to access the instance variables instead.